projects/sandbox/src/app/wizard/wizard.component.ts

```diff
@@ -0,0 +1,169 @@
+import { Component, Input } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { HttpErrorResponse } from '@angular/common/http';
+import { merge } from 'rxjs';
+import { DataPollingComponent } from '@shared';
+import { DatasetInfo, SubmissionResponseData, WizardField, WizardStep } from '../_models';
+import { SandboxService } from '../_services';
+import { ProtocolType } from '@shared';
+
+export interface ValidatorArrayHash {
+  [key: string]: Array<Validators>;
+}
+
+@Component({
+  selector: 'sb-wizard',
+  templateUrl: './wizard.component.html',
+  styleUrls: ['./wizard.component.scss']
+})
+export class WizardComponent extends DataPollingComponent {
+  error: HttpErrorResponse | undefined;
+  formProgress: FormGroup;
+  formUpload: FormGroup;
+  isBusy = false;
+  orbsHidden = true;
+  protocolType = ProtocolType;
+  progressData: DatasetInfo;
+  step: number;
+  trackId: number;
```

Comment:
Perhaps rename this so that it is clear that this is a dataset ID?

Reply:
Done